tests/Tests/Unit/ClientOptionsTraitTest.php

```diff
@@ -545,6 +546,16 @@ public function testBuildClientOptionsTwice()
         $this->assertEquals($options, $options2);
     }
 
+    public function testBuildClientOptionsWithClientOptions()
+    {
+        $client = new StubClientOptionsClient();
+        $clientOptions = new ClientOptions([]);
+        $clientOptions->setApiEndpoint('TestEndpoint.com');
+        $builtOptions = $client->buildClientOptions($clientOptions);
+
+        $this->assertEquals($clientOptions->apiEndpoint, $builtOptions['apiEndpoint']);
```

Comment:
```suggestion
        $this->assertEquals($clientOptions['apiEndpoint'], $builtOptions['apiEndpoint']);
```

Reply:
I know that we are trying to keep things compatible and such, but it breaks my heard to use array notations vs `$variable->property` </3